controllers/orano2ims_controller.go

```diff
@@ -0,0 +1,600 @@
+			return
+		}
+
+		// Create the needed ServiceAccount.
+		err = r.createServiceAccount(ctx, orano2ims, utils.ORANO2IMSDeploymentManagerServerName)
+		if err != nil {
+			r.Log.Error(err, "Failed to deploy ServiceAccount for Deployment Manager server.")
+			return
+		}
+
+		// Create the Service needed for the Metadata server.
+		err = r.createService(ctx, orano2ims, utils.ORANO2IMSDeploymentManagerServerName)
+		if err != nil {
+			r.Log.Error(err, "Failed to deploy Service for Deployment Manager server.")
+			return
+		}
+
+		// Create the metadata-server deployment.
+		err = r.deployManagerServer(ctx, orano2ims)
+		if err != nil {
+			r.Log.Error(err, "Failed to deploy the Deployment Manager server.")
+			return
+		}
+	}
+
+	err = r.updateORANO2ISMStatus(ctx, orano2ims)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("Failed to update status for ORANO2IMS %s.", orano2ims.Name))
+		nextReconcile = ctrl.Result{RequeueAfter: 30 * time.Second}
+	}
+	return
+}
+
+func (r *ORANO2IMSReconciler) deployMetadataServer(ctx context.Context, orano2ims *oranv1alpha1.ORANO2IMS) error {
+	r.Log.Info("[deployMetadataServer]")
+
+	// Build the deployment's metadata.
+	deploymentMeta := metav1.ObjectMeta{
+		Name:      utils.ORANO2IMSMetadataServerName,
+		Namespace: utils.ORANO2IMSNamespace,
+		Labels: map[string]string{
+			"oran-o2ims": orano2ims.Name,
+			"app":        utils.ORANO2IMSMetadataServerName,
+		},
+	}
+
+	// Build the deployment's spec.
+	deploymentSpec := appsv1.DeploymentSpec{
+		Replicas: k8sptr.To(int32(1)),
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"app": utils.ORANO2IMSMetadataServerName,
+			},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"app": utils.ORANO2IMSMetadataServerName,
+				},
+			},
+			Spec: corev1.PodSpec{
+				ServiceAccountName: utils.ORANO2IMSMetadataServerName,
+				Volumes: []corev1.Volume{
+					{
+						Name: "tls",
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								SecretName: fmt.Sprintf("%s-tls", utils.ORANO2IMSMetadataServerName),
+							},
+						},
+					},
+				},
+				Containers: []corev1.Container{
+					{
+						Name:            "server",
+						Image:           utils.ORANImage,
+						ImagePullPolicy: "Always",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "tls",
+								MountPath: "/secrets/tls",
+							},
+						},
+						Command: []string{"/usr/bin/oran-o2ims"},
+						Args: []string{
+							"start",
+							"metadata-server",
+							"--log-level=debug",
+							"--log-file=stdout",
+							"--api-listener-address=0.0.0.0:8000",
+							"--api-listener-tls-crt=/secrets/tls/tls.crt",
+							"--api-listener-tls-key=/secrets/tls/tls.key",
+							fmt.Sprintf("--cloud-id=%s", orano2ims.Spec.CloudId),
+							//fmt.Sprintf("--external-address=https://%s", orano2ims.Spec.IngressHost),
```

Comment:
This remind me that I need to add the `--external-address` option, thanks @irinamihai.